### PR TITLE
reality-data-react package

### PR DESCRIPTION
**Pull request by zjzoltek**
Initial implementation of the reality-data-react package. Direct port from https://dev.azure.com/bentleycs/iModelTechnologies/_git/imodeljs-ui-snippets?path=%2Fui-snippets%2Freality-data

This PR adds a new project to the ./packages folder named reality-data. This project provides a widget which allows a user to graphically access and show/hide reality data instances within an iModel.

This widget is themed using ui-core and should have no dependencies on any inner-sourced Bentley packages.

See the README on how to use the package within an application.

### CODE REVIEW

File: packages/reality-data/package.json [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  "author": {
+    "name": "Bentley Systems, Inc.",
+    "url": "http://www.bentley.com"
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/imodeljs/viewer-components-react/tree/master/packages/reality-data"
+  },
+  "peerDependencies": {
+    "@bentley/bentleyjs-core": "2.0.0",
```
Comment (aruniverse) [inline packages/reality-data/package.json:31]: peer deps shouldn't be pinned, should use `^`

File: packages/reality-data/package.json [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    "cpx": "^1.5.0",
+    "rimraf": "^3.0.2",
+    "tslint": "^5.11.0",
+    "typescript": "~3.7.4"
+  },
+  "dependencies": {
+    "classnames": "^2.2.5",
+    "i18next": "^10.2.2",
+    "react": "^16.8.0",
+    "react-dom": "^16.8.0",
```
Comment (aruniverse) [inline packages/reality-data/package.json:104]: remove react & react-dom

File: packages/reality-data/public/locales/en/IModelSelect.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{
```
Comment (aruniverse) [inline packages/reality-data/public/locales/en/IModelSelect.json:1]: I dont think you meant to add this file. This should probably be replaced with src/public/locales/en/RealityData.json

File: packages/reality-data/src/api/RealityData.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+/**
+ * RealityData class provides utility methods that must be called by host application that want RealityData support.
+ */
+export class RealityData {
+  private static _i18n?: I18N;
+  private static _complaint = "RealityData not initialized";
+
+  /** @internal */
+  public static get packageName(): string {
+    return "reality-data";
```
Comment (aruniverse) [inline packages/reality-data/src/api/RealityData.ts:16]: should be reality-data-react.

File: packages/reality-data/src/components/Widget.tsx [hunk 5]
```diff
@@ -0,0 +1,975 @@
...
+          boxSizing: "border-box",
+        }}
+      >
+        <Widget {...options} />
+      </div>
+    );
+  }
+}
+
+export default Control;
```
Comment (aruniverse) [inline packages/reality-data/src/components/Widget.tsx:973]: should prefer named exports imo. just keep it consistent through out this package

File: packages/reality-data/package.json [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    "@bentley/imodeljs-quantity": "2.0.0",
+    "@bentley/itwin-client": "2.0.0",
+    "@bentley/orbitgt-core": "2.0.0",
+    "@bentley/presentation-common": "2.0.0",
+    "@bentley/presentation-frontend": "2.0.0",
+    "@bentley/product-settings-client": "2.0.0",
+    "@bentley/ui-abstract": "2.0.0",
+    "@bentley/ui-components": "2.0.0",
+    "@bentley/ui-core": "2.0.0",
+    "@bentley/ui-framework": "2.0.0",
```
[the remark this answers is not in the document]
Comment (aruniverse) [reply packages/reality-data/package.json:49]: you beat me to it 😄

**Comment by aruniverse** [comment]
Dont forget to add code owner.

**Comment by aruniverse** [comment]
> What OIDC scopes does this package require?

def needs at least `reality-data:read`

**Comment by calebmshafer** [comment]
> > What OIDC scopes does this package require?
> 
> @calebmshafer, def needs at least `reality-data:read`

Hm okay, that makes this and the consuming react components unusable by a 3rd party...  That OIDC scope isn't support by our registration dashboard.

**Pull request closed without merging**